### PR TITLE
Feature/eicnet 1811 eu login

### PR DESCRIPTION
**Pull request by lramarojaona**
### Pre-requirements

- Add `$config['eic_user_login.settings']['endpoint_url'] = 'https://eic.test.eismea.eu/mydashboard/community/REST/get-user-json';` to your settings file
- Add `$config['eic_user_login.settings']['basic_auth_username'] = 'username';` to your settings file
- Add `$config['eic_user_login.settings']['basic_auth_password'] = 'password';`  to your settings file

### Tests

- [x] Go to `/admin/config/eic/smed` and enable both _Check/sync user_ and _Allow users to register within Drupal_
- [x] Start the mock server: `drush cas-mock-server:start`
- [x] Add following user to mock serverv: `drush casms-uc nsireste --email=demo@com.com --password=test`
- [x] As anonymous, try to login with the user just created
- [x] As SA, check that the user has been created in Drupal (this should be changed later)
- [x] As SA, edit the user in Drupal, block it, add 1 in the _SME Dashboard ID_ and put _demo@com.com_ as email
- [x] As anonymous, try to login again
- [x] Make sure you are logged in with "Welcome first_name last_name" message
- [ ] As TU, make sure you can't access `SME Dashboard ID field` when editing your user account

### Remarks

- For now we are using `cas_mock_server` module to simulate EU Login
- You will probably need to use host.docker.internal domain for cas mock server (http://host.docker.internal:8081)
- You can check existing users on your local mock server by running `drush cas-mock-server:user-list`
- We don't have access to SMED (yet?), there is only one account to test for now
- Since the SMED cannot reach local environments, we need to allow Drupal to create user accounts upon EU Login success, where normally it should be the SMED creating users
- You can find the SMED endpoint credentials here: https://citnet.tech.ec.europa.eu/CITnet/confluence/display/EICNET/WS+-+Taxonomy+webservice#WSTaxonomywebservice-Vocabularymapping under _TaxonomyWS-test_

### CODE REVIEW

File: config/sync/cas.settings.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+_core:
+  default_config_hash: W42TeamUFaYIcLMbvbx1hngcurDgrV8-Rh_iNsgeFJc
+login_link_enabled: false
+login_link_label: 'CAS Login'
+login_success_message: ''
+server:
+  version: '2.0'
+  protocol: http
+  hostname: host.docker.internal
```
Comment (Dromadoros) [inline config/sync/cas.settings.yml:9]: We need to put different parameters of this file in the config ignore no?

**Comment by Dromadoros** [comment]
As a TU I can edit my SMED id

File: config/sync/cas.settings.yml (hunk 1 above)
Comment (lramarojaona) [reply config/sync/cas.settings.yml:9]: Yes you're right, this is done.

**Comment by lramarojaona** [comment]
> As a TU I can edit my SMED id

@Dromadoros , this is fixed now.

**Review by Dromadoros — APPROVED** [review]

**Pull request merged**